### ISSUE TITLE
Unmarshalling with Dictionary issue

### ISSUE THREAD

**Issue by GearsAD** [lwabeke/Unmarshal.jl#18]
Hi,

We just upgraded to 0.1.1 and see an issue marshalling dictionaries. This may be similar to #9 ?  

Unmarshal is unable to convert the dictionary in this example:

```julia
using JSON
using Unmarshal

mutable struct TestUnmarshal
  a::String
  b::String
  links::Dict{String, String}
end

raw = "{\"a\": \"\",\"b\": \"Test\",\"links\": {\"self\": \"TestDict\"}}"
j = JSON.parse(raw)
@show val = Unmarshal.unmarshal(TestUnmarshal, j)
```

```
ERROR: Unexpected character
Line: 0
Around: ...self...
            ^

Stacktrace:
 [1] _error(::String, ::JSON.Parser.MemoryParserState) at /home/gearsad/.julia/v0.6/JSON/src/Parser.jl:140
 [2] parse_jsconstant(::JSON.Parser.MemoryParserState) at /home/gearsad/.julia/v0.6/JSON/src/Parser.jl:186
 [3] parse_value(::JSON.Parser.ParserContext{Dict{String,Any},Int64}, ::JSON.Parser.MemoryParserState) at /home/gearsad/.julia/v0.6/JSON/src/Parser.jl:170
 [4] #parse#1(::Type{Dict{String,Any}}, ::Type{Int64}, ::Function, ::String) at /home/gearsad/.julia/v0.6/JSON/src/Parser.jl:393
 [5] unmarshal(::Type{Dict{String,String}}, ::Dict{String,Any}, ::Bool, ::Int64) at /home/gearsad/.julia/v0.6/Unmarshal/src/Unmarshal.jl:143
 [6] unmarshal(::Type{T} where T, ::Dict{String,Any}, ::Bool, ::Int64) at /home/gearsad/.julia/v0.6/Unmarshal/src/Unmarshal.jl:118
 [7] unmarshal(::Type{T} where T, ::Dict{String,Any}) at /home/gearsad/.julia/v0.6/Unmarshal/src/Unmarshal.jl:94
 [8] macro expansion at /home/gearsad/.julia/v0.6/Atom/src/repl.jl:118 [inlined]
 [9] anonymous at ./<missing>:?
```
Thanks again for building this library, it's saving a lot of work not having to manually convert JSON structs!

**Comment by lwabeke** [comment]
Trying to fix this for v0.6. Not sure about the process while in this migration stage to v0.7
I hopefully tagged a release for v0.1.3 now

**Comment by lwabeke** [comment]
Julia 0.7 and 1.0 now seem to pull Unmarshal v0.2.1 by default, which should fix these issues. 
Should hopefully be no need to be pulling master anymore.
Please reopen if not the case.
